src/generic_core/remote-instance.ts

```diff
@@ -229,9 +232,12 @@ module Core {
               this.rtcToNet_ = null;
             });
           }
-          // TODO: signalFromRemote needs to get converted into a
-          // WebRtc.SignallingMessage. This probably doesn't actually work right
-          // now.
+          if (!this.rtcToNet_) {
```

Comment:
It probably is worth adding a comment to explain this. And explain that it's there to catch bugs in the underlying socks-rtc and rtc-net. Else later, someone might think that it's part of the specification of socks-rtc and rtc-net to give messages out of order.